--- a/bundler/project.py	2020-11-29 12:07:58.000000000 +0100
+++ b/bundler/project.py	2021-03-21 22:52:31.000000000 +0100
@@ -386,11 +386,11 @@
             path, fname = os.path.split(filename)
             name, ext = os.path.splitext(fname)
 
-            with open (filename, "r", encoding="utf8") as source:
+            with open (filename, "r") as source:
                 lines = source.readlines()
             gir_file = os.path.join(gir_dest, fname)
             typelib = os.path.join(typelib_dest, name + '.typelib')
-            with open (gir_file, "w", encoding="utf8") as target:
+            with open (gir_file, "w") as target:
                 for line in lines:
                     if re.match('\s*shared-library=', line):
                         (new_line, subs) = re.subn(lib_path, self.bundle_path, line)
