--- a/configure	2024-05-02 11:45:25
+++ b/configure	2024-07-13 14:35:06
@@ -10396,7 +10396,7 @@
       if ! test -d "$LIBTIFFCONFDIR" ; then
         mkdir "$LIBTIFFCONFDIR"
       fi
-      cd "$LIBTIFFCONFDIR" && "$absolute_source_path/$LIBTIFFDIR/configure" $GS_TIFF_CONFIGURE_OPTS --disable-sphinx --disable-pixarlog --disable-lerc --disable-jbig --disable-lzma --disable-zstd --disable-webp --disable-libdeflate $SUBCONFIG_OPTS
+      cd "$LIBTIFFCONFDIR" && "$LIBTIFFDIR/configure" $GS_TIFF_CONFIGURE_OPTS --disable-sphinx --disable-pixarlog --disable-lerc --disable-jbig --disable-lzma --disable-zstd --disable-webp --disable-libdeflate $SUBCONFIG_OPTS
       status=$?
       if test $status -ne 0 ; then
         as_fn_error $status "libtiff configure script failed" "$LINENO" 5
