diff --color=auto -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2023-07-07 09:39:49
+++ b/CMakeLists.txt	2023-07-15 15:58:51
@@ -94,6 +94,8 @@
 # Name of the config file used by Graphviz
 set(GVPLUGIN_CONFIG_FILE config${GRAPHVIZ_PLUGIN_VERSION})
 
+set(CMAKE_INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/${LIBRARY_INSTALL_DIR}")
+
 # ============================ Library dependencies ============================
 if(WIN32)
   if(use_win_pre_inst_libs)
diff --color=auto -Naur a/plugin/core/CMakeLists.txt b/plugin/core/CMakeLists.txt
--- a/plugin/core/CMakeLists.txt	2023-07-07 09:39:49
+++ b/plugin/core/CMakeLists.txt	2023-07-15 16:00:35
@@ -50,6 +50,8 @@
   SOVERSION ${GRAPHVIZ_PLUGIN_VERSION}
 )
 
+set_target_properties(gvplugin_core PROPERTIES INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/${PLUGIN_INSTALL_DIR}")
+
 if(MINGW)
   # work around https://gitlab.kitware.com/cmake/cmake/-/issues/21716
   set_target_properties(gvplugin_core PROPERTIES
diff --color=auto -Naur a/plugin/dot_layout/CMakeLists.txt b/plugin/dot_layout/CMakeLists.txt
--- a/plugin/dot_layout/CMakeLists.txt	2023-07-07 09:39:49
+++ b/plugin/dot_layout/CMakeLists.txt	2023-07-15 16:02:47
@@ -27,6 +27,8 @@
   ARCHIVE DESTINATION ${LIBRARY_INSTALL_DIR}
 )
 
+set_target_properties(gvplugin_dot_layout PROPERTIES INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/${PLUGIN_INSTALL_DIR}")
+
 # Specify library version and soversion
 set_target_properties(gvplugin_dot_layout PROPERTIES
   VERSION ${GRAPHVIZ_PLUGIN_VERSION}.0.0
diff --color=auto -Naur a/plugin/neato_layout/CMakeLists.txt b/plugin/neato_layout/CMakeLists.txt
--- a/plugin/neato_layout/CMakeLists.txt	2023-07-07 09:39:49
+++ b/plugin/neato_layout/CMakeLists.txt	2023-07-15 16:02:10
@@ -42,6 +42,8 @@
   ARCHIVE DESTINATION ${LIBRARY_INSTALL_DIR}
 )
 
+set_target_properties(gvplugin_neato_layout PROPERTIES INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/${PLUGIN_INSTALL_DIR}")
+
 # Specify library version and soversion
 set_target_properties(gvplugin_neato_layout PROPERTIES
   VERSION ${GRAPHVIZ_PLUGIN_VERSION}.0.0
diff --color=auto -Naur a/plugin/pango/CMakeLists.txt b/plugin/pango/CMakeLists.txt
--- a/plugin/pango/CMakeLists.txt	2023-07-07 09:39:49
+++ b/plugin/pango/CMakeLists.txt	2023-07-15 16:01:36
@@ -44,6 +44,8 @@
     ARCHIVE DESTINATION ${LIBRARY_INSTALL_DIR}
   )
 
+set_target_properties(gvplugin_pango PROPERTIES INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/${PLUGIN_INSTALL_DIR}")
+
   # Include DLLs with this library on Windows
   if(WIN32 AND install_win_dependency_dlls)
     install(
--- a/plugin/quartz/CMakeLists.txt	2023-04-30 18:35:50
+++ b/plugin/quartz/CMakeLists.txt	2023-07-20 16:43:34
@@ -38,6 +38,8 @@
     ARCHIVE DESTINATION ${LIBRARY_INSTALL_DIR}
   )

+  set_target_properties(gvplugin_quartz PROPERTIES INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/${PLUGIN_INSTALL_DIR}")
+
   set_target_properties(gvplugin_quartz PROPERTIES
     VERSION ${GRAPHVIZ_PLUGIN_VERSION}.0.0
     SOVERSION ${GRAPHVIZ_PLUGIN_VERSION}
